content/go/concepts/interfaces/interfaces.md

```diff
@@ -0,0 +1,268 @@
+
+```go
+type InterfaceName interface {
+    method() return_type
+}
+```
+
+### Example
+
+```go
+package main
+
+import (
+  "fmt"
+)
+
+type marriage interface {
+  love()
+  arranged()
+}
+
+type john struct{}
+
+func (a *john) love() {
+  fmt.Println("love with john")
+}
+
+func (a *john) arranged() {
+  fmt.Println("arranged marriage with john")
+}
+
+type max struct{}
+
+func (a *max) love() {
+  fmt.Println("love with max")
+}
+
+func (a *max) arranged() {
+  fmt.Println("arranged marriage with max")
+}
+
+func newLife(guy marriage) {
+  guy.love()
+  guy.arranged()
+}
+
+func main() {
+  newLife(&john{})
+  newLife(&max{})
+}
+```
+
+Output:
+
+```shell
+love with john
+arranged marriage with john
+love with max
+arranged marriage with max
+```
+
+In the above example:
+
+- The `marriage` interface has two methods: `love` and `arranged`.
+- `john` and `max` implement the `marriage` interface. Note that neither `john` nor `max` have any explicit declaration to implement the interface.
+- As long as the `struct` has all the methods in the interface, it is said to implement the interface.
+- The function `newLife()` takes a `marriage` interface as an argument.
+- It's possible to call `newLife()` with the `john` and `max` structures.
+
+The advantage of using an interface is that any `struct` can be passed to the `newLife()` function as long as it implements the `marriage` interface. It is possible to change the implementation of the `love` and `arranged` methods in the `john` and `max` structures without affecting the `newLife()` function.
+
+It is also possible to make a new type and `newLife()` can use it without any change as long as the new type implements the `marriage` interface.
+
+Go's compiler will ensure that all the methods in the interface are implemented in the `struct`.
+
+A type implements an interface by implementing its methods, In `Go`, interfaces are implemented implicitly and enables polymorphism.
+
+## Empty Interface
+
+An interface with no methods is called an empty interface.
+
+```go
+type interfaceName interface {}
+```
+
+The tradeoff of using an empty interface is the loss of type safety. Since there are no methods, it's possible to pass any type to a function that takes an empty interface as an argument.
+
+### Example
+
+The `Println()` function in the `fmt` package takes an empty interface as an argument.
+
+Below is the Go source code for the `fmt.Println()` function:
+
+```go
+// Println formats using the default formats for its operands and writes to standard output.
+// Spaces are always added between operands and a newline is appended.
+// It returns the number of bytes written and any write error encountered.
+func Println(a ...interface{}) (n int, err error) {
+    return Fprintln(os.Stdout, a...)
+}
+```
+
+> **Note:** Due to the empty interface, any type can be passed to the `Println()` function.
+
+## Embedding an Interface
+
+An interface in go can be embedded in other interfaces or structs.
```

Comment:
@yangc95  one of the benefit for the approach i said at https://github.com/Codecademy/docs/pull/1708#discussion_r1162217190 
would be as you can see in the below example i just put embeding interface in anthr interface n struct in 1 single example we can divide this into two diff examples ....